src/Core/Resolvers/SqlMutationEngine.cs

```diff
@@ -1073,14 +1085,116 @@ private void PopulateParamsFromRestRequest(Dictionary<string, object?> parameter
+                    );
+                }
+            }
+            
+            return fieldsToAuthorize;
+        }
+
+        private void PopulateMutationFieldsToAuthorize(
+            Dictionary<string, HashSet<string>> fieldsToAuthorize,
+            IInputField schema,
+            string entityName,
+            IMiddlewareContext context,
+            object? parameters,
+            RuntimeConfig runtimeConfig)
+        {
+            InputObjectType schemaObject = ResolverMiddleware.InputObjectTypeFromIInputField(schema);
+            if (!fieldsToAuthorize.ContainsKey(entityName))
+            {
+                fieldsToAuthorize.Add(entityName, new HashSet<string>());
+            }
+
+            if (parameters is List<ObjectFieldNode> fields)
+            {
+                ProcessObjectFieldNodesForAuthZ(fieldsToAuthorize, entityName, context, runtimeConfig, schemaObject, fields);
             }
+            else if (parameters is ObjectValueNode objectValue)
+            {
+                ProcessObjectFieldNodesForAuthZ(fieldsToAuthorize, entityName, context, runtimeConfig, schemaObject, objectValue.Fields);
+            }
+            else if (parameters is List<IValueNode> values)
+            {
+                values.ForEach(value => PopulateMutationFieldsToAuthorize(fieldsToAuthorize, schema, entityName, context, value, runtimeConfig));
+            }
+        }
+
+        private void ProcessObjectFieldNodesForAuthZ(Dictionary<string, HashSet<string>> fieldsToAuthorize, string entityName, IMiddlewareContext context, RuntimeConfig runtimeConfig, InputObjectType schemaObject, IReadOnlyList<ObjectFieldNode> fieldNodes)
+        {
+            foreach (ObjectFieldNode field in fieldNodes)
+            {
+                Tuple<IValueNode?, SyntaxKind> fieldDetails = GetUnderlyingKindForField(field.Value, context.Variables);
+                SyntaxKind underlyingFieldKind = fieldDetails.Item2;
+                if (underlyingFieldKind != SyntaxKind.ObjectValue && underlyingFieldKind != SyntaxKind.ListValue)
+                {
+                    fieldsToAuthorize[entityName].Add(field.Name.Value);
+                }
+                else
+                {
+                    string relationshipName = field.Name.Value;
+                    string targetEntityName = runtimeConfig.Entities![entityName].Relationships![relationshipName].TargetEntity;
+                    PopulateMutationFieldsToAuthorize(
+                        fieldsToAuthorize,
+                        schemaObject.Fields[relationshipName],
+                        targetEntityName,
+                        context,
+                        fieldDetails.Item1,
+                        runtimeConfig);
+                }
+            }
+        }
+
+        private Tuple<IValueNode?, SyntaxKind> GetUnderlyingKindForField(IValueNode? value, IVariableValueCollection variables)
```

Comment:
Can you elaborate on what is "kind" and the significance ?

Reply:
a name more descriptive than `value` would be helpful here.

Reply:
added method summary